lib/puppet_x/bodeco/util.rb

```diff
@@ -4,7 +4,7 @@ module PuppetX
   module Bodeco
     module Util
       def self.download(url, filepath, options = {})
-        uri = URI(url)
+        uri = Puppet::Util.path_to_uri(uri) unless uri.is_a?(URI)
```

Comment:
The spec is failing here because you should use url instead of uri, otherwise url is an undeclared variable:
Puppet::Util.path_to_uri(url)
instead of:
Puppet::Util.path_to_uri(uri)